crates/db_perf/src/main.rs

```diff
@@ -0,0 +1,174 @@
+use clap::Parser;
+use diesel::{
+  dsl::{self, sql},
+  sql_types,
+  ExpressionMethods,
+  IntoSql,
+};
+use diesel_async::{RunQueryDsl, SimpleAsyncConnection};
+use lemmy_db_schema::{
+  schema::post,
+  source::{
+    community::{Community, CommunityInsertForm},
+    instance::Instance,
+    person::{Person, PersonInsertForm},
+  },
+  traits::Crud,
+  utils::{
+    build_db_pool,
+    get_conn,
+    now,
+    series::{self, ValuesFromSeries},
+  },
+  SortType,
+};
+use lemmy_db_views::{post_view::PostQuery, structs::PaginationCursor};
+use lemmy_utils::error::LemmyResult;
+use std::num::NonZeroU32;
+
+#[derive(Parser, Debug)]
+struct CmdArgs {
+  #[arg(long, default_value_t = 3.try_into().unwrap())]
+  communities: NonZeroU32,
+  #[arg(long, default_value_t = 3.try_into().unwrap())]
+  people: NonZeroU32,
+  #[arg(long, default_value_t = 100000.try_into().unwrap())]
+  posts: NonZeroU32,
+  #[arg(long, default_value_t = 0)]
+  read_post_pages: u32,
+  #[arg(long)]
+  explain_insertions: bool,
+}
+
+#[tokio::main]
+async fn main() -> LemmyResult<()> {
+  if let Err(err) = try_main().await {
+    println!("😂 Error: {err:?}");
+  }
+  if let Ok(path) = std::env::var("PGDATA") {
+    println!("🪵 query plans and error details written in {path}/log");
+  }
+
+  Ok(())
+}
```

Comment:
I dont see the purpose of this wrapper, Rust should handle errors from main just fine.

Reply:
Also can you run this code from CI with low number of posts and communities to make sure it doesnt break in the future?

Reply:
The custom error handling makes it possible to show the log path when an error happens. I now changed it to be cleaner and return the correct exit code.

I also made it run in CI.